x-pack/elastic-agent/pkg/agent/cmd/setup_config.go

```diff
@@ -0,0 +1,119 @@
+	Port                string              `config:"port"`
+	CustomHeaders       map[string]string   `config:"headers"`
+	CustomKibanaHeaders map[string]string   `config:"kibana_headers"`
+}
+
+type elasticsearchConfig struct {
+	CA           string `config:"ca"`
+	Host         string `config:"host"`
+	Username     string `config:"username"`
+	Password     string `config:"password"`
+	ServiceToken string `config:"service_token"`
+}
+
+type kibanaConfig struct {
+	Fleet              kibanaFleetConfig `config:"fleet"`
+	RetrySleepDuration time.Duration     `config:"retry_sleep_duration"`
+	RetryMaxCount      int               `config:"retry_max_count"`
+}
+
+type kibanaFleetConfig struct {
+	CA       string `config:"ca"`
+	Host     string `config:"host"`
+	Password string `config:"password"`
+	Setup    bool   `config:"setup"`
+	Username string `config:"username"`
+}
+
+func defaultAccessConfig() (setupConfig, error) {
+	retrySleepDuration, err := envDurationWithDefault(defaultRequestRetrySleep, requestRetrySleepEnv)
+	if err != nil {
+		return setupConfig{}, err
+	}
+
+	retryMaxCount, err := envIntWithDefault(defaultMaxRequestRetries, maxRequestRetriesEnv)
+	if err != nil {
+		return setupConfig{}, err
+	}
+
+	cfg := setupConfig{
+		Fleet: fleetConfig{
+			CA:              envWithDefault("", "FLEET_CA", "KIBANA_CA", "ELASTICSEARCH_CA"),
+			Enroll:          envBool("FLEET_ENROLL", "FLEET_SERVER_ENABLE"),
+			EnrollmentToken: envWithDefault("", "FLEET_ENROLLMENT_TOKEN"),
+			Force:           envBool("FLEET_FORCE"),
+			Insecure:        envBool("FLEET_INSECURE"),
+			TokenName:       envWithDefault("Default", "FLEET_TOKEN_NAME"),
+			TokenPolicyName: envWithDefault("", "FLEET_TOKEN_POLICY_NAME"),
+			URL:             envWithDefault("", "FLEET_URL"),
+		},
+		FleetServer: fleetServerConfig{
+			Cert:    envWithDefault("", "FLEET_SERVER_CERT"),
+			CertKey: envWithDefault("", "FLEET_SERVER_CERT_KEY"),
+			Elasticsearch: elasticsearchConfig{
+				Host:         envWithDefault("http://elasticsearch:9200", "FLEET_SERVER_ELASTICSEARCH_HOST", "ELASTICSEARCH_HOST"),
+				Username:     envWithDefault("elastic", "FLEET_SERVER_ELASTICSEARCH_USERNAME", "ELASTICSEARCH_USERNAME"),
+				Password:     envWithDefault("changeme", "FLEET_SERVER_ELASTICSEARCH_PASSWORD", "ELASTICSEARCH_PASSWORD"),
+				ServiceToken: envWithDefault("", "FLEET_SERVER_SERVICE_TOKEN"),
+				CA:           envWithDefault("", "FLEET_SERVER_ELASTICSEARCH_CA", "ELASTICSEARCH_CA"),
+			},
+			Enable:              envBool("FLEET_SERVER_ENABLE"),
+			Host:                envWithDefault("", "FLEET_SERVER_HOST"),
+			InsecureHTTP:        envBool("FLEET_SERVER_INSECURE_HTTP"),
+			PolicyID:            envWithDefault("", "FLEET_SERVER_POLICY_ID", "FLEET_SERVER_POLICY"),
+			Port:                envWithDefault("", "FLEET_SERVER_PORT"),
+			CustomHeaders:       envMap("FLEET_CUSTOM_HEADER"),
+			CustomKibanaHeaders: envMap("FLEET_CUSTOM_KBN_HEADER"),
```

Comment:
Why `KBN` here? we don't use that shortcut anywhere else in the ENV's, I would stick with `FLEET_KIBANA_HEADER`.